File: 백준/Bronze/2231. 분해합/분해합.py
import sys

n = int(sys.stdin.readline())

for i in range(n-54, n):
    tmp = i
    check = 0
    
    while tmp > 0:
        check += tmp % 10
        tmp //= 10
    
    if check + i == n:
        sys.stdout.write(f"{i}\n")
        break
else:
    sys.stdout.write(f"0\n")

File: 백준/Bronze/2231. 분해합/분해합.cc
#include <iostream>
using namespace std;

int main() {
    int n;
    bool check = false;
    cin >> n;

    for (int i = n-54; i < n; i++) {
        int tmp = i, num = 0;
        while (tmp > 0) {
            num += tmp % 10;
            tmp /= 10;
        }

        if (num + i == n) {
            cout << i << "\n";
            check = true;
            break;
        }
    }

    if (!check)
        cout << 0 << "\n";
    
    return 0;
}